test/jdk/java/net/httpclient/ExpectContinueResponseMessageTest.java

```diff
@@ -0,0 +1,186 @@
+                    InputStream inputStream = socket.getInputStream();
+                    InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+
+                    StringBuilder stringBuilder = new StringBuilder();
+
+                    // Read initial request
+                    byte b;
+                    while (true) {
+                        b = (byte) inputStreamReader.read();
+                        stringBuilder.append((char) b);
+
+                        if (stringBuilder.length() >= 4) {
+                            char[] lastBytes = new char[4];
+                            stringBuilder.getChars(
+                                    stringBuilder.length() - 4,
+                                    stringBuilder.length(), lastBytes, 0);
+                            if (Arrays.equals(lastBytes, new char[]{'\r', '\n', '\r', '\n'})) {
+                                break;
+                            }
+                        }
+                    }
+                    OutputStream outputStream = socket.getOutputStream();
+
+                    //send a wrong response
+                    outputStream.write(control.statusLine.getBytes());
+                    outputStream.flush();
+                } catch (SocketException e) {
+                    // ignore
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        serverThread = new Thread(runnable);
+        serverThread.start();
+    }
+
+    @AfterAll
+    public void stopServerSocket() throws Exception {
+        Control control = this.control;
+        control.stop = true;
+        control.serverSocket.close();
+        serverThread.join();
+    }
+
+    @ParameterizedTest
+    @MethodSource("args")
+    public void test(int expectedCode, String statusLine, String expectedMessage) throws Exception {
+        String body = "Testing: " + expectedCode;
+        Control control = this.control;
+        control.statusLine = statusLine + "\r\n\r\n";
+
+        HttpURLConnection connection = createConnection();
+        connection.setFixedLengthStreamingMode(body.getBytes().length);
+        try {
+            OutputStream outputStream = connection.getOutputStream();
+            outputStream.write(body.getBytes());
+            outputStream.close();
+        } catch (Exception ex) {
+            // swallow the exception
+        }
+
+        int responseCode = connection.getResponseCode();
+        String responseMessage = connection.getResponseMessage();
+        assertTrue(responseCode == expectedCode,
+                String.format("Expected %s response, instead received %s", expectedCode, responseCode));
+        assertTrue(expectedMessage.equals(responseMessage),
+                String.format("Expected Response Message  %s, instead received %s",
+                        expectedMessage, responseMessage));
```

Comment:
maybe the connection should be explicitely closed at the end.
maybe the accepted socket should be added to Control and closed here too?